lib/rules/prefer-optional-props-using-with-defaults.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * @author @neferqiqi
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+/**
+ * @typedef {import('../utils').ComponentTypeProp} ComponentTypeProp
+ */
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+// ...
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'enforce props with default values ​​to be optional',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/prefer-optional-props-using-with-defaults.html'
+    },
+    fixable: 'code',
+    schema: [],
+    messages: {
+      // ...
+    }
+  },
+  /** @param {RuleContext} context */
+  create(context) {
+    /**
+     * @param {ComponentTypeProp} prop
+     * @param {Token[]} tokens
+     * */
+    const findKeyToken = (prop, tokens) =>
+      tokens.find((token) => {
+        const isKeyIdentifierEqual =
+          prop.key.type === 'Identifier' && token.value === prop.key.name
+        const isKeyLiteralEqual =
+          prop.key.type === 'Literal' && token.value === prop.key.raw
+        return isKeyIdentifierEqual || isKeyLiteralEqual
+      })
+
+    return utils.defineScriptSetupVisitor(context, {
+      onDefinePropsEnter(node, props) {
+        if (!utils.hasWithDefaults(node)) {
+          return
+        }
+        const withDefaultsProps = Object.keys(
+          utils.getWithDefaultsPropExpressions(node)
+        )
+        const requiredProps = props.flatMap((item) =>
+          item.type === 'type' && item.required ? [item] : []
+        )
+
+        for (const prop of requiredProps) {
+          if (withDefaultsProps.includes(prop.propName)) {
+            const firstToken = context.getSourceCode().getFirstToken(prop.node)
+            // skip setter & getter case
+            if (firstToken.value === 'get' || firstToken.value === 'set') {
+              return
+            }
+            // skip computed
+            if (prop.node.computed) {
+              return
+            }
+            const keyToken = findKeyToken(
+              prop,
+              context.getSourceCode().getTokens(prop.node)
+            )
+            if (!keyToken) return
+            context.report({
+              node: prop.node,
+              loc: prop.node.loc,
+              data: {
+                key: prop.propName
+              },
+              message: `Prop "{{ key }}" should be optional.`,
+              fix: (fixer) => fixer.insertTextAfter(keyToken, '?')
```

Comment:
We don't know if it's right to make it optional or to remove the default, so I think it makes more sense to use the suggestion API to provide suggestions instead of autofix.